### ISSUE TITLE
`on-remove` doesn't work with single select and `allow-clear`.

### ISSUE THREAD

**Issue by antch** [angular-ui/ui-select#1225]
It appears that currently the `onRemove` callback is only executed when a multiselect option is removed. It makes sense that it should also be called when the user clears a single select when `allow-clear="true"`.

**Comment by bradgreens** [comment]
:+1:

**Comment by damjtoh** [comment]
+1

**Comment by martynchamberlin** [comment]
+1

**Comment by wesleycho** [comment]
Can you post a Plunker reproducing this?

Commenters: Please avoid using +1, and use GitHub reactions - +1s only serve to pollute issues, preventing people from finding the important information needed to craft a fix/analyze a problem.

**Comment by martynchamberlin** [comment]
Sorry. GitHub Reactions came out literally a couple days after my comment, and long before the other +1s. But we'll definitely remember to use those going forward!

**Comment by antch** [comment]
http://plnkr.co/edit/ah66gkgU5TxTKnXFw9NF?p=preview

**Comment by edwin177** [comment]
For now, it looks like on-select is called with $item = null.

**Comment by jigarakatidus** [comment]
@edwin177 on-select looks fine to me. issue is with on-remove as of now. http://plnkr.co/edit/Y8RskftrBB659cyG7M7K?p=preview

**Comment by edwin177** [comment]
Yup, I ran into the same issue and found this thread. I was just sharing my work-around until it's fixed. When you click clear, on-select is called with $item = null.

**Comment by antch** [comment]
Actually, `on-select` is called with `undefined`, not `null`.

**Comment by RodrigoFraga** [comment]
+1

**Comment by iberodev** [comment]
any progress on this?

**Comment by PratibhaP** [comment]
I think, currently on-remove works only for multiple select.
So, we need to continue with on-select event with an undefined check.

**Comment by Jefiozie** [comment]
Indeed the on-remove only works on the multiple select, see link.
[multiple select on-remove](https://github.com/angular-ui/ui-select/blob/master/src/uiSelectMultipleDirective.js#L54)

Maybe somebody can have a look at the root cause and make a purpose for this issue?

**Comment by kuuurt13** [comment]
Ran into this issue today and created a PR (#1977). This will add the same multi select onRemove functionality to single selects with `allow-clear="true"`

**Comment by awaneeshd** [comment]
@kuuurt13 is this fix merged , i still face the same issue
